Backend-API/src/main/java/com/avangers/backendapi/services/UserServiceImpl.java

```diff
@@ -35,4 +35,22 @@ public RegisterUserResponseDTO addUser(RegisterUserRequestDTO registerUserReques
   public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
     return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException("Email is not in database"));
   }
+
+  @Override //updating the user data
+  public RegisterUserResponseDTO updateUser(Long userId, RegisterUserRequestDTO updateUserRequestDTO) {
+    User existingUser = userRepository.findById(userId)
+            .orElseThrow(() -> new IllegalArgumentException("User not found"));
```

Comment:
UsernameNotFoundException?

Reply:
Oh, yes that would be way better, I was just going with the recommendations of IntelliJ/copilot. You are correct, or we can write one of our own.